Libraries/MiscDrivers/Touchscreen/tsc2007.c

```diff
@@ -0,0 +1,330 @@
+ * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL MAXIM INTEGRATED BE LIABLE FOR ANY CLAIM, DAMAGES
+ * OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * Except as contained in this notice, the name of Maxim Integrated
+ * Products, Inc. shall not be used except as stated in the Maxim Integrated
+ * Products, Inc. Branding Policy.
+ *
+ * The mere transfer of this software does not imply any licenses
+ * of trade secrets, proprietary technology, copyrights, patents,
+ * trademarks, maskwork rights, or any other form of intellectual
+ * property whatsoever. Maxim Integrated Products, Inc. retains all
+ * ownership rights.
+ *
+ ******************************************************************************
+ *
+ * Copyright 2023 Analog Devices, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************************/
+
+#include <stddef.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <string.h>
+#include <stdio.h>
+
+#include "mxc_device.h"
+#include "mxc_delay.h"
+#include "tsc2007.h"
+
+/************************************ DEFINES ********************************/
+#ifndef FLIP_SCREEN
+#define FLIP_SCREEN 0
+#endif
+#ifndef SWAP_XY
+#define SWAP_XY 0
+#endif
+#ifndef ROTATE_SCREEN
+#define ROTATE_SCREEN 0
+#endif
+#define X_RES_T 320
+#define Y_RES_T 240
+
+#define ADC_X_MIN 232
+#define ADC_X_MAX 3888
+#define ADC_Y_MIN 375
+#define ADC_Y_MAX 3799
+
+
+/******************************* TYPE DEFINITIONS ****************************/
+typedef struct _TS_Buttons_t {
+    int x0;
+    int y0;
+    int x1;
+    int y1;
+    int key_code;
+} TS_Buttons_t;
+
+static TS_Buttons_t ts_buttons[TS_MAX_BUTTONS];
+static int pressed_key = 0;
+static mxc_i2c_regs_t *t_i2c;
+static unsigned int t_i2c_freq;
+static mxc_gpio_cfg_t int_gpio;
+static mxc_gpio_cfg_t t_i2c_gpio;
+unsigned int g_x, g_y = 0;
+// Global touchscreen event flag that can be polled by applications if needed.
+// The application should clear it to 0 if used.
+// It is set to 1 eve
+int ts_event = false;
+
+uint16_t tsX, tsY, tsZ1;
+
+static uint8_t tsConstructCommand(mxc_ts_cmd_func_t function, mxc_ts_cmd_pdown_t pdown, mxc_ts_cmd_mode_t mode)
```

Comment:
Should you use camel case or snake case for function names?

Reply:
camelCase used for function names of static functions.